pkg/tsdb/grafana-postgresql-datasource/postgres.go

```diff
@@ -104,20 +104,25 @@ func newPostgres(ctx context.Context, cfg *setting.Cfg, dsInfo sqleng.DataSource
 	return db, handler, nil
 }
 
-func (s *Service) newInstanceSettings(cfg *setting.Cfg) datasource.InstanceFactoryFunc {
+func (s *Service) newInstanceSettings() datasource.InstanceFactoryFunc {
 	logger := s.logger
 	return func(ctx context.Context, settings backend.DataSourceInstanceSettings) (instancemgmt.Instance, error) {
-		logger.Debug("Creating Postgres query endpoint")
+		cfg := backend.GrafanaConfigFromContext(ctx)
+		sqlCfg, err := cfg.SQL()
+		if err != nil {
+			return nil, err
+		}
+
 		jsonData := sqleng.JsonData{
-			MaxOpenConns:        cfg.SqlDatasourceMaxOpenConnsDefault,
-			MaxIdleConns:        cfg.SqlDatasourceMaxIdleConnsDefault,
-			ConnMaxLifetime:     cfg.SqlDatasourceMaxConnLifetimeDefault,
+			MaxOpenConns:        sqlCfg.DefaultMaxOpenConns,
+			MaxIdleConns:        sqlCfg.DefaultMaxOpenConns,
```

Comment:
Incorrect value is set here, shouldn't this be `sqlCfg.DefaultMaxIdleConns`

Reply:
yes, you're correct, nice find 👍 . fixed.